src/main/java/org/jboss/arquillian/testcontainers/TestContainersObserver.java

```diff
@@ -49,21 +50,35 @@ public void stopContainer(@Observes AfterClass afterClass) {
 
     private void checkForDocker(boolean failIfNoDocker) {
         final String detailMessage = "No Docker environment is available.";
-        if (!isDockerAvailable() && failIfNoDocker) {
-            throw new AssertionError(detailMessage);
-        } else {
-            try {
-                Class<?> clazz = Class.forName("org.junit.AssumptionViolatedException");
-                Constructor<?> ctor = clazz.getDeclaredConstructor(String.class);
-                throw (RuntimeException) ctor.newInstance(detailMessage);
-            } catch (Exception e) {
-                // TestNG?
+        if (!isDockerAvailable()) {
+            if (failIfNoDocker) {
+                throw new AssertionError(detailMessage);
+            } else if (!isDockerAvailable()) {
+                // First attempt to throw a JUnit 5 assumption
+                try {
+                    Class<?> clazz = Class.forName("org.opentest4j.TestAbortedException");
+                    Constructor<?> ctor = clazz.getConstructor(String.class);
+                    throw (RuntimeException) ctor.newInstance(detailMessage);
+                } catch (ClassNotFoundException | NoSuchMethodException | InstantiationException | IllegalAccessException
+                        | InvocationTargetException ignore) {
+                }
+                // Not found, attempt to throw a JUnit exception
+                try {
+                    Class<?> clazz = Class.forName("org.junit.AssumptionViolatedException");
+                    Constructor<?> ctor = clazz.getDeclaredConstructor(String.class);
+                    throw (RuntimeException) ctor.newInstance(detailMessage);
+                } catch (ClassNotFoundException | NoSuchMethodException | InstantiationException | IllegalAccessException
+                        | InvocationTargetException ignore) {
+                }
```

Comment:
Extract to a method and pass in the exception class as a String to deduplicate?

Reply:
+1 good idea.